.github/workflows/shared-create-hotfix-pr.yml

```diff
@@ -0,0 +1,180 @@
+        required: true
+        type: string
+      main-tag-regex:
+        description: "the regex to find all related main releases"
+        required: true
+        type: string
+      changelog-path:
+        description: "path to the changelog file"
+        required: false
+        default: ./CHANGELOG.md
+        type: string
+      changelog-config-path:
+        description: "path to the changelog config"
+        required: true
+        type: string
+      pr-body-text:
+        description: "text to be included in the PR"
+        required: false
+        type: string
+      release-target:
+        description: "target of the release (rust|wasm)"
+        required: true
+        type: string
+    secrets:
+      GPG_PRIVATE_KEY:
+        description: "GPG private key for signing commits and tags"
+        required: true
+      GPG_PASSPHRASE:
+        description: "GPG private passphrase for signing commits and tags"
+        required: true
+        
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          # Number of commits to fetch. 0 indicates all history for all branches and tags.
+          fetch-depth: 0
+          ref: ${{ inputs.branch }}
+
+      - name: Determine Hotfix Version
+        run: |
+
+          TAG_PREFIX=${{ inputs.tag-prefix }}
+          BRANCHNAME=${{ inputs.branch }}
+
+          SUPPORT_TAG_BASE=${BRANCHNAME##*/}
+
+          ALL_TAGS=$(git tag --sort=-version:refname -l)
+          temp_array=()
+          for value in ${ALL_TAGS[@]}
+          do
+              # consider tags matching the RegEx for main tags and matching the support branch base
+              if [[ $value =~ ${{ inputs.main-tag-regex }} && $value == $SUPPORT_TAG_BASE* ]]; then
+                LATEST_TAG=$value
+                break
+              fi
+          done
+          unset temp_array
+          echo LATEST_TAG=$LATEST_TAG
+
+          # takes a string and increments the last number
+          # e.g. $(updateVersion 1.1.0) -> 1.1.1
+          updateVersion()
+          {
+            [[ ${1} =~ ^(.*[^0-9])?([0-9]+)$ ]]  && \
+              [[ ${#BASH_REMATCH[1]} -gt 0 ]] && \
+                printf "%s%0${#BASH_REMATCH[2]}d" "${BASH_REMATCH[1]}" "$((10#${BASH_REMATCH[2]} + 1 ))" || \
+                printf "%0${#BASH_REMATCH[2]}d" "$((10#${BASH_REMATCH[2]} + 1))" || \
+              printf "${1}"
+          }
+
+          HOTFIX_TAG=$(updateVersion $LATEST_TAG)
+          echo HOTFIX_TAG=$HOTFIX_TAG
+          echo HOTFIX_TAG=$HOTFIX_TAG >> $GITHUB_ENV
+
+          HOTFIX_VERSION=$(echo "$HOTFIX_TAG" | sed "s/$TAG_PREFIX*//")
+          echo HOTFIX_VERSION=$HOTFIX_VERSION
+          echo HOTFIX_VERSION=$HOTFIX_VERSION >> $GITHUB_ENV
+
+      - name: Determine Excluded Tags
+        run: |
+
+          # joins an array with a delimiter
+          function join_by { local IFS="$1"; shift; echo "$*"; }
```

Comment:
So much of this bash code is duplicated across three workflows now. It would be nice either to make a shared action or generalise the code and put in `scripts/`.